Enumerable features

helciodev opened the pull request:
## This feature includes the following methods:
- #my_each
- #my_each_with_index
- #my_select
- #my_all?
- #my_any?
- #my_none?
- #my_count
- #my_map
- #my_inject
- multiply_els method that uses my_inject Enumerable method

helciodev committed:
adds initial files

helciodev committed:
adds my_each to module Enumerable

helciodev committed:
adds my_each_with_index to module Enumerable

helciodev committed:
adds my_select to module Enumerable

helciodev committed:
adds my_all? to module Enumerable

helciodev committed:
adds my_any? to module Enumerable

helciodev committed:
adds my_none? to module Enumerable

helciodev committed:
adds my_count to module Enumerable

helciodev committed:
adds my_map to module Enumerable

helciodev committed:
adds my_inject to module Enumerable

helciodev committed:
adds multiply_els method to module Enumerable

helciodev committed:
adds disanbling comment

adds the following comments on the top:
rubocop:disable Metrics/CyclomaticComplexity
rubocop: disable Metrics/PerceivedComplexity
rubocop: disable Metrics/MethodLength
rubocop: disable Style/For

helciodev committed:
updates README file

helciodev committed:
updates README file

helciodev committed:
corrects errors

helciodev committed:
fixes errors

helciodev committed:
fixes errors

ebeagusamuel reviewed (changes requested):
# Changes Requested :recycle: 
hello @helciodev  :wave: 
You have done nice work so far on this project.
Well done :clap: 
There are still some improvements to be made as there are some test cases some of your methods are failing. Before I begin highlighting them, I would advise that after writing your custom implementation of the enumerable methods, you test them against a different range of input with which you will also test the original methods to see if they return the same result. The goal is to make your methods behave exactly like the original methods such that when passed the same arguments, they both return the same result.
- [x] Your my_each and my_each_with_index methods fail when tested against a range or hash. The original method is able to work on both range and hashes, so please update your method to be able to do the same.
- [x] Just as stated above, your my_select method should be able to work on a range.
- [x] Your my_any? and my_none? methods are not behaving the way the original behaves. It should be able to work on ranges too. I would advise you to check the ruby docs for these methods and see exactly how they work and what they return when tested against different ranges of inputs and arguments and try to make your own methods behave like that.
- [x] There is an error in your implementation of the my_count method. It doesn't account for some test cases. Currently, it is only able to handle cases where a block is being passed as an argument. The original count methods can handle more than just a case where a block is passed to it. As stated above, I highly suggest you visit the ruby docs to study these enumerable methods and see how they really work.
- [x] Your my_map and my_inject should be able to work on a range just like the original.

Please request another review when you are done making the requested changes.
slack(@ebeagusamuel)

helciodev committed:
adds correct names

helciodev committed:
adds changes required after TSE feedback

method my_each now works for not only arrays but for hashes and ranges as well, same as for all the other methods

ElijahTobs reviewed (changes requested):
# Status: Changes Required :recycle: 

### Hi @helciodev 
Kudos :+1: You have done well so far.
However, there are some changes you need to make before the project can be approved.

Some of your custom enumerable methods failed some test cases. 

- [ ] Your my_each and my_each_with_index methods fail when tested against a range
- [ ] Your my_all and my_any methods also failed. Make sure you test them against  different ranges of inputs alongside the original methods and be sure they both return the same output.
- [ ] The my_count method also failed.

### Request another code review when done making the required changes

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

helciodev committed:
changes applied after TSE review

all methods now work how they are supposed to

Srjnnnn reviewed (approved):
### Status: Approved :partying_face: :+1: 
Hello team. Congratulations your project has been approved. Thanks for making the requested changes from previous TSEs. Your all methods passed the test cases.

There are some optional changes that can be made.

- [ ] You can improve your Readme. It looks good now. However, you can specify your codes on the readme by putting them in backticks. It'll look better and more professional like that.

- [ ] It would be good to write more descriptive commit messages and commit more often.

Congrats again. :smile: Good luck with the next projects.
Eyüp Sercan UYGUR
